crates/sui-faucet/src/metrics_layer.rs

```diff
@@ -69,18 +72,19 @@ where
         self.inner.poll_ready(ctx)
     }
 
-    fn call(&mut self, req: Req) -> Self::Future {
-        let metrics = MetricsGuard::new(self.metrics.clone());
+    fn call(&mut self, req: Request<Body>) -> Self::Future {
+        let path = req.uri().path().to_string();
+        let metrics = MetricsGuard::new(self.metrics.clone(), &path);
         let inner = self.inner.clone();
 
         let future = Box::pin(async move {
             let resp = inner.oneshot(req).await;
             match &resp {
-                Result::Ok(resp) if !resp.status().is_success() => {
+                Ok(resp) if !resp.status().is_success() => {
                     metrics.failed(None, Some(resp.status()))
                 }
-                Result::Ok(_) => metrics.succeeded(),
-                Result::Err(err) => {
+                Ok(_) => metrics.succeeded(),
```

Comment:
what happens if status code is 429? does it go through the Ok(_) branch or through Err branch?

Reply:
429 will go through Ok(Resp) branch and match the if condition, and go to metrics.failed

Reply:
ah yes, I misread line 79-83, I thought the if statement was removed. All good!